src/jobs/boost_rewards.rs

```diff
@@ -0,0 +1,300 @@
+            .ok_or_else(|| anyhow::anyhow!("Earn vault address not configured"))?;
+        let earn_vault_addr = Address::from_str(earn_vault_address)?;
+
+        if self.dry_run {
+            println!("✅ DRY RUN: Would transfer {} {} to Earn Vault", 
+                     daily_amount_human, token_symbol);
+            println!("✅ DRY RUN: Would call onBoostReward({}, {})", 
+                     self.token_address, daily_amount_wei);
+            return Ok(());
+        }
+
+        // 7. Transfer tokens to Earn Vault
+        println!("📤 Transferring tokens to Earn Vault...");
+        let transfer_tx = execute_with_retry(
+            || {
+                let contract = token_contract.clone();
+                let amount = daily_amount_wei;
+                let to = earn_vault_addr;
+                async move { contract.transfer(to, amount).await }
+            },
+            &retry_config,
+            "Token transfer",
+        ).await?;
+
+        println!("   Transfer TX: {:?}", transfer_tx);
+
+        // Monitor transfer transaction
+        let timeout_gas_used = U256::from_str(&self.config.monitoring.timeout_gas_used)?;
+        let monitor = TransactionMonitor::new_with_timeout_values(
+            client.provider(),
+            Duration::from_secs(self.config.monitoring.transaction_timeout_seconds),
+            Duration::from_secs(self.config.monitoring.poll_interval_seconds),
+            self.config.monitoring.timeout_block_number,
+            timeout_gas_used,
+        );
+
+        let transfer_receipt = monitor.monitor_transaction(transfer_tx).await?;
+        match transfer_receipt.status {
+            TransactionStatus::Success => {
+                println!("✅ Transfer confirmed in block {}", transfer_receipt.block_number);
+            }
+            TransactionStatus::Failed => {
+                return Err(anyhow::anyhow!("Token transfer failed"));
+            }
+            TransactionStatus::Timeout => {
+                return Err(anyhow::anyhow!("Token transfer monitoring timeout"));
+            }
+        }
+
+        // 8. Call onBoostReward
+        println!("📞 Calling onBoostReward on Earn Vault...");
+        let earn_vault = EarnVaultContract::new(
+            earn_vault_addr,
+            client.provider(),
+            client.clone(),
+        );
+
+        let boost_reward_tx = execute_with_retry(
+            || {
+                let contract = earn_vault.clone();
+                let token = self.token_address;
+                let amount = daily_amount_wei;
+                async move { contract.on_boost_reward(token, amount).await }
+            },
+            &retry_config,
+            "onBoostReward call",
+        ).await?;
+
+        println!("   onBoostReward TX: {:?}", boost_reward_tx);
+
+        // Monitor onBoostReward transaction
+        let boost_reward_receipt = monitor.monitor_transaction(boost_reward_tx).await?;
```

Comment:
all monitor transactions implementation can be replaced by `with_timeout` method of `provider` as seen here https://docs.rs/alloy/latest/alloy/providers/struct.PendingTransactionBuilder.html#examples